### ISSUE TITLE
Terminal: clear command history is a no-op

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#143701]
Testing #143430
Linux

1. Run some commands in your terminal
2. F1 > Terminal Clear Command History
3. F1 > Terminal run recent commands
4. All the commands are still there, nothing got cleared 🐛 

![Screenshot 2022-02-23 at 11 56 00](https://user-images.githubusercontent.com/1926584/155306002-2aa15950-3bab-4e05-b98b-e22c35df167c.png)

**Comment by Tyriar** [comment]
This is by design as clearing history only clears the previous session commands. We don't discard the current session as they're all in the active terminal marked by decorations. It does however clear the active terminal's current session history from the previous session history (which hides by default as they would duplicate the content).

**Comment by isidorn** [comment]
I find this very counterintuitive... As no other clear command in VS Code behaves that way.

**Comment by Tyriar** [comment]
I cleared recently opened and the active windows still show up, this is essentially the same thing the terminal history is doing:

![image](https://user-images.githubusercontent.com/2193314/155355587-caf2c149-fdcd-45f3-a95a-5b84fd5c44f8.png)

**Comment by paulacamargo25** [comment]
On macOS, i run `F1 > Terminal Clear Command History`, and when i run `Terminal run recent commands` again, i can see all the commands, including those from the previous session.
<img width="802" alt="Screen Shot 2022-02-24 at 11 07 24 AM" src="https://user-images.githubusercontent.com/17892325/155591074-9d42f95c-afb4-4347-bbe1-94a60b327c07.png">

**Comment by meganrogge** [comment]
I agree that it's a little confusing. Perhaps we should say `clear previous session history` instead?

**Comment by rzhao271** [comment]
I now get the clear command history and a clear previous session command history command. Neither command seems to be clearing the command history.
I'm on Windows with pwsh.

![Screencap of me running both commands, but neither seem to be clearing the command history.](https://user-images.githubusercontent.com/7199958/214711363-897da1ce-d1b7-409e-96db-ec1036683e03.gif)

**Comment by Tyriar** [comment]
@meganrogge the change you made was to the quickaccess code, this is nothing to do with the terminal so we need to revert that

**Comment by isidorn** [comment]
@Tyriar can you help with verification steps for this one. Since I do not see the "Clear Command History" in quick pick. Does this only appear on Linux?

**Comment by meganrogge** [comment]
@isidorn the title of the command is `Terminal: Clear Previous Session History`

**Comment by isidorn** [comment]
@meganrogge thanks. After running that commend I still see the recent commands. So the command history is not cleared. Is this expected?

**Comment by rzhao271** [comment]
I still see previous session history commands given to the terminal after running the `Terminal: Clear Previous Session History` command.

**Comment by Tyriar** [comment]
@rzhao271 is the group called  `previous session` or is it `pwsh history` or something similar? Only the first is expected to be cleared.
